x/gov/handler.go

```diff
@@ -179,6 +187,31 @@ func (h TallyHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx)
 	if err := proposal.Tally(); err != nil {
 		return nil, err
 	}
+	if proposal.Result == Proposal_Accepted {
+		switch proposal.Type {
+		case Proposal_Text: //nothing to do
+		case Proposal_UpdateElectorate:
+			// hard coded here, revisit and maybe store in a map with callbacks
+			details := proposal.GetElectorateUpdateDetails()
+			if details == nil {
+				return nil, errors.Wrap(errors.ErrHuman, "details not set")
+			}
+			// merge
```

Comment:
I think it would be better to write comment explaining **why** merge. This and the next comment (_merge without validation_) does not provide any real information. I can see from the code what is executed. I cannot see **why** it is written like this.

Reply:
Good feedback, will do!